src/charm.py

```diff
@@ -561,29 +575,75 @@ def _on_mongod_pebble_ready(self, event) -> None:
 
         except (PathError, ProtocolError, MissingSecretError) as e:
             logger.error("Cannot initialize workload: %r", e)
-            event.defer()
-            return
+            return False
 
-        # Add initial Pebble config layer using the Pebble API
-        container.add_layer("mongod", self._mongod_layer, combine=True)
-        container.add_layer("log_rotate", self._log_rotate_layer, combine=True)
+        return True
+
+    def __configure_layers(self, container: Container) -> bool:
```

Comment:
where possible, we should try to avoid functions that do something and those that return something.

Reply:
Lost my answer somewhere but:
 * The code in this function is supposed to make the caller defer and return eagerly, so I need to propagate the information.
 * I can replace it by a try/catch but it will be same/same from the caller perspective: catch an exception/`false` means defer + return, otherwise go on.
 What do you prefer?

Reply:
> I can replace it by a try/catch but it will be same/same from the caller perspective: catch an exception/false means defer + return, otherwise go on.

would be great! Thank you very much!